src/main/scala/com/gu/support/workers/GetRecurringSubscription.scala

```diff
@@ -0,0 +1,54 @@
+package com.gu.support.workers
+
+import cats.implicits._
+import com.gu.support.workers.lambdas.IdentityId
+import com.gu.support.workers.model.AccessScope.{AccessScopeBySessionId, AccessScopeNoRestriction, SessionId}
+import com.gu.support.workers.model.{AccessScope, BillingPeriod}
+import com.gu.zuora.GetAccountForIdentity.DomainAccount
+import com.gu.zuora.GetSubscription.DomainSubscription
+import com.gu.zuora.ZuoraConfig.RatePlanId
+import com.gu.zuora.ZuoraService
+import com.gu.zuora.model.response.RatePlan
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object GetRecurringSubscription {
+
+  def apply(
+    zuoraService: ZuoraService,
+    accessScope: AccessScope,
+    identityId: IdentityId,
+    billingPeriod: BillingPeriod
+  )(implicit ec: ExecutionContext): Future[Option[DomainSubscription]] = {
+
+    val productRatePlanId = Option(zuoraService.config).map(_.contributionConfig(billingPeriod).productRatePlanId)
+
+    def hasContributorPlan(sub: DomainSubscription): Boolean =
+      productRatePlanId match {
+        case Some(productRatePlanId) => GetRecurringSubscription.hasContributorPlan(productRatePlanId, sub.ratePlans)
+        case None => false
+      }
+
+    def isInScope(domainAccount: DomainAccount): Boolean =
```

Comment:
Could we possibly get a comment describing what `isInScope` means? Maybe it's just me but I'm finding it a little hard to folow

Reply:
I have separated this into a function `IsSubscriptionAccessAllowed` and switched from using Option to an equivalent named trait ExistingContributionSessionId.  I feel like this is much clearer, but have a look and see if it can be improved more.

Reply:
I have to say I'm with Jonny, I feel like scope is an odd name for this. What does scope have to do with determining whether a user has already made a contribution or not?

Reply:
yes agreed I'm not sure if you can see the latest commit but I have changed it to say `isAccountAccessAllowed` @rupertbates 
There is still one thing called AccessScope but that's in support-models and I was too lazy to rename and do a PR there.  Happy to do so if people would like though.